lib/cadet/assessments/assessments.ex

```diff
@@ -787,14 +835,70 @@ defmodule Cadet.Assessments do
     end
   end
 
+  defp find_team(assessment_id, cr_id)
+       when is_ecto_id(assessment_id) and is_ecto_id(cr_id) do
+    query =
+      from(t in Team,
+        where: t.assessment_id == ^assessment_id,
+        join: tm in assoc(t, :team_members),
+        where: tm.student_id == ^cr_id,
+        limit: 1
+      )
+
+    assessment_team_size =
+      Map.get(
+        Repo.one(
+          from(a in Assessment,
+            where: a.id == ^assessment_id,
+            select: %{max_team_size: a.max_team_size}
+          )
+        ),
+        :max_team_size,
+        0
+      )
+
+    case assessment_team_size > 1 do
+      true ->
+        case Repo.one(query) do
+          nil -> {:error, :team_not_found}
+          team -> {:ok, team}
+        end
+
+      # team is nil for individual assessments
+      false ->
+        {:ok, nil}
+    end
+  end
+
   def get_submission(assessment_id, %CourseRegistration{id: cr_id})
       when is_ecto_id(assessment_id) do
-    Submission
-    |> where(assessment_id: ^assessment_id)
-    |> where(student_id: ^cr_id)
-    |> join(:inner, [s], a in assoc(s, :assessment))
-    |> preload([_, a], assessment: a)
-    |> Repo.one()
+    query =
+      from(t in Team,
+        where: t.assessment_id == ^assessment_id,
+        join: tm in assoc(t, :team_members),
+        where: tm.student_id == ^cr_id,
+        limit: 1
+      )
+
+    team = Repo.one(query)
+
+    case team do
```

Comment:
This might be a source of bug, please confirm again. You can have a team assessment, but if the team is not formed yet, the `team` will be `nil` and it assumes that it's an individual submission.

The check should still be like `find_team` above where you first check the `max_team_size` of the assessment to determine whether it's a team assessment or an individual assessment.

Reply:
This is not an issue. The student will only be able to trigger this part of the code after finalizing submission. In the context of Team Submission, they should already have a team formed if they reach this juncture. Even if the team is deleted right as the student submits, there is a clause handling nil submission in the assessments controller.